File: Main/static/JS/game/effects/slowdown.js
import { Modificator } from "../classes/modificator.js";
import { DurableEffect } from "./effect.js";


class Slowdown extends DurableEffect {
    constructor(hero, game_state, duration, caster, value){
        super('замедление', 'slowdown.png', hero, game_state, duration, caster)
        this.value=value
    }

    gen_description(){
        return `Замедление[${this.value}]  (длительность = ${this.duration}) (наложено: ${this.caster.name})`
    }

    before_move(){
        this.decrease_duration()
    }

    gen_modificators(){
        return [new Modificator('slowdown', 1, this.value, 'after', 'sum')]
    }
}

export {Slowdown}

File: Main/models/effects/slowdown.py
from .effect import DurableEffect
from django.db import models
from Main.models.modifier import Modifier


class Slowdown(DurableEffect):
    name = 'Slowdown'
    img_src = 'slowdown.png'

    value = models.IntegerField()

    def gen_description(self, link):
        return f"Замедление ({self.value} на {self.duration} ходов) (наложено {link.caster.name})"

    def before_move(self, link):
        self.decrease_duration(link)

    def gen_modifiers(self, link):
        return [Modifier(hero=link.hero, param_name='slowdown', value=self.value, duration=1)]
